public/systems_monitoring_graphs.js

```diff
@@ -0,0 +1,1087 @@
+            let bisectDate = d3.bisector(function(d) {return d.datetime;}).left;
+        
+            function drawFocus() {    
+                // Create focus object
+                let focus = svg.append("g")
+                    .attr("class", "focus")
+    
+                // Append circle on the line path
+                focus.append("circle")
+                    .attr("r", 3.5)
+    
+                // Add background rectangle behind the text tooltip
+                focus.append("rect")
+                    .attr("x", -30)
+                    .attr("y", "-32px")
+                    .attr("rx", 6)
+                    .attr("ry", 6)
+                    .attr("width", 220)
+                    .attr("height", 20);
+    
+                // Add text annotation for tooltip
+                focus.append("text")
+                    .attr("x", -20)
+                    .attr("dy", "-18px")
+                    .attr("font-size", "12px")
+                    .style("fill", "#0E86D4");
+    
+                // Create an overlay path to draw the above objects on top of
+                areaChart
+                    .attr("class", "overlay")  
+                    .on("mouseover", () => focus.style("display", null))
+                    .on("mouseout", () => focus.style("display", "none"))
+                    .on("mousemove", tipMove);
+
+                // Make the overlay rectangle transparent,
+                // So it only serves the purpose of detecting mouse events
+                d3.select(".overlay")
+                    .style("fill", "none")
+                    .style("pointer-events", "all");
+    
+                // Select focus objects and set opacity
+                d3.selectAll(".focus")
+                    .style("opacity", 0.9);
+
+                // Select the circle and style it
+                d3.selectAll(".focus circle")
+                    .style("fill", "#068ca0")
+                    .style("opacity", 0)
+
+                // Select the rect and style it
+                d3.selectAll(".focus rect")
+                    .style("fill", "whitesmoke")
+                    .style("opacity", 0)
+    
+                // Function that adds tooltip on hover
+                function tipMove() {
+                    // Below code finds the date by bisecting and
+                    // Stores the x and y coordinate as variables
+                    let x0 = x.invert(d3.mouse(this)[0]);
+                    let i = bisectDate(data, x0, 1);
+                    let d0 = data[i - 1];
+                    let d1 = data[i];
+                    let d = x0 - d0.datetime > d1.datetime - x0 ? d1 : d0;
+    
+                    // Place the focus objects on the same path as the line
+                    focus.attr("transform", `translate(${x(d.datetime)}, ${y(d.disk_usage.used)})`);
+    
+                    // Position the text
+                    focus.select("text")
```

Comment:
Can you update this to show the nearest data point (i.e. data for the point highlighted by the circle) instead of the interpolated timestamps and values?

Reply:
Updated, thanks.